### ISSUE TITLE
Veyon does not work over two different Bullseye releases

### ISSUE THREAD

**Issue by ralf-krause** [puavo-org/puavo-os#609]
I activated the veyon tool for classroom management to test it in my Puavo school machmitnetz.einszueins.opinsys.fi.

My Puavo device pool002.einszueins.opinsys.fi should be the teacher device and the Puavo devices pool001, pool003, pool004 and pool005 should be the student devices. I must update all devices to the same Puavo release opinsys-bullseye-2022-10-06-105238-amd64 to get a working Veyon connection. 

Also the Puavo configuration parameters changed with the from opinsys-os-opinsys-bullseye-2022-09-08-105038-amd64 to opinsys-bullseye-2022-10-06-105238-amd64. I don't really know how the parameters were for the older image ... okay, Veyon is confgured for the BBS Beims in Magdeburg so I can see the parameters set to the student devices. I got the following info from the school's setting.

  "puavo.service.puavo-veyon.enabled": "true",
  "puavo.service.veyon.enabled": "true",
  "puavo.veyon.role": "slave",

I can't say if the Veyon software is implemented in the image and all schools will get it automatically or if you need to add the following configuration parameter for the school or for the complete organization. I can't see this because I'm only admin in the BBS Beims and I am not an owner of the organisation magdeburg.

  "puavo.pkg.veyon": "latest",

I could not find the Puavo configuration parameter for the teacher role in the Magdeburg administration website. Does the school really know how to activate Veyon in the teacher role?  I also could not find any hint in the https://images.opinsys.fi/meta/opinsys-os-opinsys-bullseye-amd64.json for the older images.

For the newer Puavo image you need different configuration parameters to activate Veyon roles on the Puavo devices. The student roll needs the Puavo configuration parameter in the last line of the following parameters.

  "puavo.pkg.veyon": "latest",
  "puavo.service.puavo-veyon.enabled": "true",
  "puavo.service.veyon.enabled": "true",
  "puavo.veyon.role_rules": ":client"

For the teacher role I found the following parameter.

  "puavo.veyon.role_rules": "%teacher:master",

I can't say how we will get an update from from opinsys-os-opinsys-bullseye-2022-09-08-105038-amd64 to opinsys-bullseye-2022-10-06-105238-amd64 without having problems with Veyon in the BBS Beims. Is there a useful way to update the Puavo devices?

**Comment by juhaerk** [comment]
Our Veyon integration had many changes in the latest release, and I did not try to preserve compatibility with the old version. This breakage was unavoidable I think, or not worth the effort. From now on I will try to keep the client-master negotiation protocol backwards compatible, so this incompatibility should not happen in the future except perhaps on some exceptional circumstances.

The "puavo.veyon.role" puavo-conf is no longer used. "puavo.veyon.role_rules" replaces this and is more flexible. The default setting for it is "%admin,%teacher:master :client", meaning that admins and teachers get the master functionality and everyone else gets the client functionality. The syntax is a list of "group,group,...:role" items, where %admin, %teacher, %student are possible "virtual-groups" meaning user role, and role can be "master", "client" or "master+client". Group can be empty in which case all users are included (including "guest"-user).

**Comment by ralf-krause** [comment]
Hey Juha,
does this mean that any person who is teacher in a school automatically gets the Veyon teacher dashboard? And it doesn't matter which Puavo device she is using?

**Comment by juhaerk** [comment]
Yes. This is only on hosts where Veyon is enabled (puavo.service.puavo-veyon.enabled and puavo.service.veyon.enabled must be true). If this policy is not good the puavo-conf "puavo.veyon.role_rules" can be tweaked on organisation, school or device level. It should also be possible to set up a specific "veyon"-group (or some other name) in Puavo and use that in puavo-conf, but I have not tested that.

Another note: the puavo.pkg.veyon is no longer needed and I think we should remove that as an option (veyon deb-packages are installed into the system image and they contain our custom patches).

**Comment by ralf-krause** [comment]
Great work! I setup the Puavo configuration for my demo school machmitnetz.einszueins.opinsys.fi for all devices ... yes, the same thing for all.

  "puavo.veyon.role_rules": "%admin,%teacher:master :client"

I started one device with a teacher account and another with 4 different users ... some of them are students and one another teacher. I got the correct Veyon roles to connect to an existing Veyon teacher by pin number or open a teacher dashboard on a second device.

The new Veyon version is very nice. At the moment only 17 devices are running the latest Puavo image and 183 do not in the BBS Beims. And if they want to use the new Veyon they must help update their devices in a fast process. This will help to get Veyon running. The schools must know about the new Veyon and the need of this update.

Thank you for your detailed answers. It is very useful to know all about Veyon for me because I will have some workshops for the teachers in Magdeburg ... and I think I will do it also for the support team in the school administration in Magdeburg.
